esphome/components/touchscreen/binary_sensor/touchscreen_binary_sensor.cpp

```diff
@@ -10,15 +10,13 @@ void TouchscreenBinarySensor::setup() {
 
 void TouchscreenBinarySensor::touch(TouchPoint tp) {
   bool touched = (tp.x >= this->x_min_ && tp.x <= this->x_max_ && tp.y >= this->y_min_ && tp.y <= this->y_max_);
-
-  if (this->page_ != nullptr) {
+  if ((this->page_ != nullptr) && (this->parent_->get_display() != nullptr)) {
```

Comment:
The display cant be null with these changes.
```suggestion

  if (this->page_ != nullptr) {
```